framework/test_framework.py

```diff
@@ -5,10 +5,154 @@
+    """
+    tree = Devicetree.parseFile(file_path)
+    test_config['platform'] = \
+        tree.children[0].properties[0].values[0]
+    test_config['nix_file'] = \
+        tree.children[0].children[0].children[0].properties[0].values[0]
+    test_config['suites'] = \
+        tree.children[0].children[0].children[0].properties[1].values[0]
+    test_config['tests'] = \
+        tree.children[0].children[0].children[0].properties[2].values[0]
+    test_config['tests_configs']['log_level'] = \
+        tree.children[0].children[0].children[0].properties[2].values[0]
+
+def run_command_in_terminal(command):
+    """
+    Run a command in a new Terminal window.
+
+    Args:
+        command (str): The command to execute.
+    """
+    # pylint: disable=R1732
+    terminal_process = subprocess.Popen(
+        ['/bin/bash', '-c', command],
+        stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE
+    )
+    # pylint: enable=R1732
+
+    return terminal_process
+
+def terminate_children_processes(parent_process):
+    """
+    Terminate all child processes of the given parent process.
+
+    Args:
+        parent_process: The parent process whose children will be terminated.
+    """
+    parent = psutil.Process(parent_process.pid)
+    children = parent.children(recursive=True)
+    for child in children:
+        child.terminate()
+        child.wait()
+
+def deploy_test(platform):
+    """
+    Deploy a test on a specific platform.
+
+    Args:
+        platform (str): The platform to deploy the test on.
+    """
+    if platform in ["qemu-aarch64-virt", "qemu-riscv64-virt"]:
+        arch = platform.split("-")[1]
+        run_cmd = "../platform/qemu/run.sh " + arch
+        process = run_command_in_terminal(run_cmd)
+        # Connection to platform will run here
+        # which will stall the platform untill the results
+        # are sent from the platform to the framework
+        terminate_children_processes(process)
 
 if __name__ == '__main__':
 
     print(cons.BLUE_TEXT + "Framework init..." + cons.RESET_COLOR)
+    print(cons.BLUE_TEXT +
+          "Framework init..." +
+          cons.RESET_COLOR)
+
+    args = PARSER.parse_args()
+
+    print(cons.BLUE_TEXT +
+          "Reading config.dts..." +
+          cons.RESET_COLOR)
+
+    dts_path = args.dts_path
+    print("config.dts file: " + dts_path)
+
+    if args.dts_path is None:
+        print(cons.RED_TEXT +
+              "Error: Please provide the --dts_path argument." +
+              cons.RESET_COLOR)
+    else:
+        dts_path = args.dts_path
+
+    parse_dts_file(dts_path)
+    print(cons.GREEN_TEXT + "config.dts successfully read!" + cons.RESET_COLOR)
+
+    print(cons.BLUE_TEXT +
+          "Creating tests source file..." +
+          cons.RESET_COLOR)
+    CURR_DIR = os.getcwd()
+    print("CURR_DIR: " + CURR_DIR)
+    os.chdir("../")
+    RUN_CMD = "python3 codegen.py -dir ../src/ "

```

Comment:
Now you are assuming a specific structure (`bao-test` at the same level as `src` with the tests inside it). However, this is merely a reference structure/example that we provide. It may change in the future, or the user may want to follow another structure. I think the `-dir` should be able to receive a generic tests dir.

Reply:
Totally agree! Commit https://github.com/bao-project/bao-tests/pull/11/commits/a497b6ccd9cc1a2bde8152f7abc63630073ab03d resolves this issue. Thank you for your feedbackl